release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py

```diff
@@ -0,0 +1,484 @@
+    # a fraction of images, then we should actually call .limit(), but right now
+    # this materializes all data to the object store. For now, we can just skip
+    # the call since we're generating all the data files to have the right
+    # number of images anyway.
+    # See https://github.com/ray-project/ray/issues/29122.
+    # ds = ray.data.read_tfrecords(filenames).limit(num_images_per_epoch)
+    return ds
+
+
+FIELDS = [
+        "data_loader",
+        "train_sleep_time_ms",
+        "num_epochs",
+        "num_images_per_epoch",
+        "num_images_per_input_file",
+        "batch_size",
+        "shuffle_buffer_size",
+        "ray_mem_monitor_enabled",
+        "ray_spilled_mb",
+        "ray_restored_mb",
+        "min_available_mb",
+        "time_total_s",
+        "tput_images_per_s",
+    ]
+
+def write_metrics(data_loader, command_args, metrics, output_file):
+    row = {key: val for key, val in metrics.items() if key in FIELDS}
+    row["data_loader"] = data_loader
+    for field in FIELDS:
+        val = getattr(command_args, field, None)
+        if val is not None:
+            row[field] = val
+
+    if "tput_images_per_s" in metrics:
+        row["tput_images_per_s"] = metrics["tput_images_per_s"]
+    else:
+        row["tput_images_per_s"] = args.num_images_per_epoch * args.num_epochs / metrics["time_total_s"]
+
+    for field in FIELDS:
+        print(f"{field}: {row[field]}")
+
+    write_header = not os.path.exists(output_file)
+    with open(output_file, 'a+', newline='') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=FIELDS)
+        if write_header:
+            writer.writeheader()
+        writer.writerow(row)
+
+    test_output_json_envvar = "TEST_OUTPUT_JSON"
+    test_output_json_path = os.environ.get(test_output_json_envvar)
+    if not test_output_json_path:
+        print(f"Environment variable {test_output_json_envvar} not set, will not write test output json.")
+    else:
+        print(f"Environment variable {test_output_json_envvar} set to '{test_output_json_path}'. Will write test output json.")
+        append_to_test_output_json(test_output_json_path, row)
+
+def append_to_test_output_json(path, metrics):
+    
+    output_json = {}
+    try:
+        with open(path, 'r') as existing_test_output_file:
+            output_json = json.load(existing_test_output_file)
+    except FileNotFoundError:
+        pass
+
+    runs = output_json.get("runs", [])
+    runs.append(metrics)
+    output_json["runs"] = runs
+    
+    num_images_per_file = metrics["num_images_per_input_file"]
+    data_loader = metrics["data_loader"]
+
+    perf_metrics = output_json.get("perf_metrics", [])
+    perf_metrics.append({
+        "perf_metric_name": f"{data_loader}_{num_images_per_file}-images-per-file_throughput-img-per-second",
+        "perf_metric_value": metrics["tput_images_per_s"],
+        "perf_metric_type": "THROUGHPUT",
+    })
```

Comment:
@jjyao should confirm but I think you may also have to set "time" and "success" as top-level JSON fields.

For "success", probably best to check whether an exception was thrown in the main training loop (`exc is not None`).

Reply:
Those are not required.

Reply:
Ah okay I thought success would get reported to the nightly dashboard, but I guess not! Yeah it's fine if we don't add duration.